### DISCUSSION TITLE
Multiple commands in raw sql migrations

### DISCUSSION THREAD

**Discussion by halldorrt** [SeaQL/sea-orm#1322] [Q&A] [answered]
Hi, 

My team and I would like to write our migrations is sql. Sea-orm offers that feature, however the migration tool only seems to be able to execute one command at a time. If we perform more than one command in the same sql string we get the following error: `Execution Error: error returned from database: cannot insert multiple commands into a prepared statement`

I can get around it by using the DO command in postgres:
```sql
DO $$ BEGIN
  -- All the statements you could ever dream of
END $$;
```

I also came across this: https://github.com/SeaQL/sea-query/discussions/437
This is slightly different however, since it revolves around executing sql files. We are writing the sql in the migration.

I am just wondering whether there is some better way that sea-orm provides of doing this, and whether multiple commands will be supported without a workaround in the future?

**Comment by billy1624** [comment, marked as answer]
Hey @halldorrt, welcome and thanks for asking!! I think we can provide an interface to execute an unprepared statement(s). I'll try to draft a PR for this later today.

> Since this is more related to sea-orm, I'd transfer this discussion.

**Reply by billy1624** [reply]
Hey @halldorrt, please check https://github.com/SeaQL/sea-orm/pull/1327

**Reply by halldorrt** [reply]
Hi, sorry for the late response, the holidays :)
Thanks a lot, I will check that out.

**Comment by tyt2y3** [comment]
You can also call `get_connection` and basically do whatever you need
